src/catalogue.js

```diff
@@ -118,26 +109,29 @@ function requestDateDataSets(projectId, dateDataSetsRequest) {
 }
 
 export function loadDateDataSets(projectId, options) {
-    let bucketItems = get(cloneDeep(options), 'bucketItems.buckets');
-
-    if (bucketItems) {
-        bucketItems = bucketItemsToExecConfig(bucketItems, { removeDateItems: true });
-    }
-
-    const omittedOptions = ['filter', 'types', 'paging', 'dataSetIdentifier', 'returnAllDateDataSets', 'returnAllRelatedDateDataSets'];
-    // includeObjectsWithTags has higher priority than excludeObjectsWithTags,
-    // so when present omit excludeObjectsWithTags
-    if (options.includeObjectsWithTags) {
-        omittedOptions.push('excludeObjectsWithTags');
+    const mdObj = get(cloneDeep(options), 'bucketItems');
+    let bucketItemsPromise;
+    if (mdObj) {
+        bucketItemsPromise = bucketItemsToExecConfig(projectId, mdObj, { removeDateItems: true, attributesMap: get(options, 'attributesMap') });
+    } else {
+        bucketItemsPromise = Promise.resolve();
```

Comment:
Check if the following code works with Promise.resolve([nothing]). What about the `then` part? Eventually we could move the check for mdObj validity to the `bucketItemsToExecConfig` method

Reply:
it works, covered by UT